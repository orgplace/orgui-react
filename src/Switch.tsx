import classNames from 'classnames';
import React from 'react';

interface OuiSwitchAttributes {
  checked?: boolean;
  onChange?: React.ChangeEventHandler<HTMLInputElement>;
}

const OuiSwitch: React.FC<React.HTMLAttributes<HTMLElement> & OuiSwitchAttributes> = (props) => {
  const {checked, onChange, ...labelProps} = props;
  return (
    <label {...labelProps} className={classNames(
      'ouiSwitch',
      props.className,
    )}>
      <input type="checkbox" className="ouiSwitch__input" checked={checked} onChange={onChange} />
      <span className="ouiSwitch__slider"></span>
    </label>
  );
}

export { OuiSwitch };
