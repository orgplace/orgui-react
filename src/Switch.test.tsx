import React from 'react'
import { OuiSwitch } from './Switch'
import renderer from 'react-test-renderer';

test('OuiSwitch', function() {
	const component = renderer.create(<OuiSwitch>test</OuiSwitch>);

	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
