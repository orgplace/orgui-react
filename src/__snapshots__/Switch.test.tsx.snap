// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OuiSwitch 1`] = `
<label
  className="ouiSwitch"
>
  <input
    className="ouiSwitch__input"
    type="checkbox"
  />
  <span
    className="ouiSwitch__slider"
  />
</label>
`;
